src/app/modules/shared-components/components/textarea/textarea.component.ts

```diff
@@ -1,16 +1,28 @@
-import { Component, OnInit, Input } from '@angular/core';
+import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
 
 @Component({
   selector: 'alg-textarea',
   templateUrl: './textarea.component.html',
   styleUrls: ['./textarea.component.scss'],
 })
 export class TextareaComponent implements OnInit {
-  @Input() icon: string;
-  @Input() placeholder: string;
-  @Input() value: string;
+  @Input() icon = '';
+  @Input() placeholder = '';
+
+  @Output() valueChange = new EventEmitter<string>();
+
+  value: string;
 
   constructor() {}
 
   ngOnInit() {}
+
+  setValue(text: string) {
+    this.value = text;
+    this.onChange();
```

Comment:
`this.value = text;` seems to call already onChange...

Reply:
I did some testing and it wasn't the case, so I kept it like this.